src/Features/Core/Portable/CodeRefactorings/NamedArguments/AbstractAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.NamedArguments
+{
+    internal abstract class AbstractAddNamedArgumentsCodeRefactoringProvider<TArgumentSyntax> : CodeRefactoringProvider
+    {
+        public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var textSpan = context.Span;
+            if (textSpan.IsEmpty)
```

Comment:
I'm not sure about this.  I could imagine just moving my cursor into some invocation and wanting to say "add named arguments".  Having to select means having to know precisely what range you need to select.  That makes sense for things like extraction and selecting multiple things to operate on.  It makes less sense for fixing up an affected symbol.  For example "change signature" does not require you to select.  it just reqiures you to be on the signature in question